src/accountsdb/fuzz.zig

```diff
@@ -45,56 +49,93 @@ pub fn run(seed: u64, args: *std.process.ArgIterator) !void {
+    defer test_data_dir.close();
+
+    const snapshot_dir_name = "accountsdb_fuzz";
+    var snapshot_dir = try test_data_dir.makeOpenPath(snapshot_dir_name, .{});
+    defer snapshot_dir.close();
     defer {
         // NOTE: sometimes this can take a long time so we print when we start and finish
         std.debug.print("deleting snapshot dir...\n", .{});
-        std.fs.cwd().deleteTree(snapshot_dir) catch |err| {
-            std.debug.print("failed to delete snapshot dir: {}\n", .{err});
+        test_data_dir.deleteTree(snapshot_dir_name) catch |err| {
+            std.debug.print("failed to delete snapshot dir ('{s}'): {}\n", .{ sig.utils.fmt.tryRealPath(snapshot_dir, "."), err });
         };
         std.debug.print("deleted snapshot dir\n", .{});
     }
     std.debug.print("use disk: {}\n", .{use_disk});
 
-    var accounts_db = try AccountsDB.init(allocator, logger, .{
+    const alt_dir_name = "alt";
+    var alt_dir = try snapshot_dir.makeOpenPath(alt_dir_name, .{});
+    defer alt_dir.close();
+    defer {
+        // NOTE: sometimes this can take a long time so we print when we start and finish
+        std.debug.print("deleting snapshot dir...\n", .{});
+        test_data_dir.deleteTree(alt_dir_name) catch |err| {
+            std.debug.print("failed to delete snapshot dir ('{s}'): {}\n", .{ sig.utils.fmt.tryRealPath(snapshot_dir, "."), err });
+        };
+        std.debug.print("deleted snapshot dir\n", .{});
+    }
+
+    var accounts_db = try AccountsDB.init(gpa, logger, snapshot_dir, .{
         .use_disk_index = use_disk,
-        .snapshot_dir = snapshot_dir,
         // TODO: other things we can fuzz (number of bins, ...)
     });
     defer accounts_db.deinit(true);
 
-    const exit = try allocator.create(std.atomic.Value(bool));
+    const exit = try gpa.create(std.atomic.Value(bool));
+    defer gpa.destroy(exit);
     exit.* = std.atomic.Value(bool).init(false);
 
     const manager_handle = try std.Thread.spawn(.{}, AccountsDB.runManagerLoop, .{
         &accounts_db,
         exit,
     });
-
-    var tracked_accounts = std.AutoArrayHashMap(Pubkey, TrackedAccount).init(allocator);
-    defer {
-        for (tracked_accounts.keys()) |key| {
-            tracked_accounts.getEntry(key).?.value_ptr.deinit(allocator);
-        }
-        tracked_accounts.deinit();
+    errdefer {
+        exit.store(true, .seq_cst);
+        manager_handle.join();
     }
+
+    var tracked_accounts = std.AutoArrayHashMap(Pubkey, TrackedAccount).init(gpa);
+    defer tracked_accounts.deinit();
+    defer for (tracked_accounts.values()) |*value| {
+        value.deinit(gpa);
+    };
     try tracked_accounts.ensureTotalCapacity(10_000);
 
-    var largest_rooted_slot: usize = 0;
-    var slot: usize = 0;
+    var random_bank_fields = try BankFields.random(gpa, rand, 1 << 8);
+    defer random_bank_fields.deinit(gpa);
 
-    const Actions = enum { put, get };
+    const random_bank_hash_info: BankHashInfo = .{
```

Comment:
probably want to move this to a `.random()` method too

Reply:
c3ef4fcf472d1faefc0f5aa0137f146425334b63